### PR TITLE
test(connman): include test in the syncsslipemaps

### PR DESCRIPTION
**Pull request by danyecls**

### CODE REVIEW

File: pkg/connman/syncslicemap_test.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.title, func(t *testing.T) {
+			ssm := tc.setup()
+			result, ok := ssm.Load(tc.key)
+
+			if ok != tc.expected.expectedOk || result != tc.expected.expectedResult {
+				t.Errorf("Test case '%s' failed. Expected (result, ok): (%v, %v), but got: (%v, %v)", tc.title, tc.expected.expectedResult, tc.expected.expectedOk, result, ok)
+			}
```
[the remark this answers is not in the document]
Comment (heiytor) [reply pkg/connman/syncslicemap_test.go:54]: I also believe that the `expected` prefix inside the `Expected` struct is redundant.

File: pkg/connman/syncslicemap_test.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+	for _, tc := range cases {
+		t.Run(tc.title, func(t *testing.T) {
+			ssm := tc.setup()
+			ssm.Store(tc.key, tc.value)
+
+			result, ok := ssm.syncMap.Load(tc.key)
+			if !ok || result == nil {
+				t.Errorf("Failed to store value '%v' for key '%s'", tc.value, tc.key)
+			}
```
Comment (heiytor) [inline pkg/connman/syncslicemap_test.go:94]: Here too

File: pkg/connman/syncslicemap_test.go [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+	for _, tc := range cases {
+		t.Run(tc.title, func(t *testing.T) {
+			ssm := tc.setup()
+			ssm.Delete(tc.key, tc.valueToDelete)
+
+			result, ok := ssm.syncMap.Load(tc.key)
+			if !reflect.DeepEqual(result, tc.expected.expectedValues) || ok != tc.expected.expectedOk {
+				t.Errorf("Test case '%s' failed. Expected (values, ok): (%v, %v), but got: (%v, %v)", tc.title, tc.expected.expectedValues, tc.expected.expectedOk, result, ok)
+			}
```
Comment (heiytor) [inline pkg/connman/syncslicemap_test.go:139]: Here

File: pkg/connman/syncslicemap_test.go [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.title, func(t *testing.T) {
+			ssm := tc.setup()
+			size := ssm.Size(tc.key)
+
+			if size != tc.expected.expectedSize {
+				t.Errorf("Test case '%s' failed. Expected size: %d, but got size: %d", tc.title, tc.expected.expectedSize, size)
+			}
```
Comment (heiytor) [inline pkg/connman/syncslicemap_test.go:202]: And here

**Review by heiytor — CHANGES_REQUESTED** [review]

File: pkg/connman/syncslicemap_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (danyecls) [reply pkg/connman/syncslicemap_test.go]: changed from t.Error to assert and renamed Expected prefix to 'expectedStatus'

File: pkg/connman/syncslicemap_test.go (hunk 2 above)
Comment (danyecls) [reply pkg/connman/syncslicemap_test.go]: changed to

File: pkg/connman/syncslicemap_test.go (hunk 4 above)
Comment (danyecls) [reply pkg/connman/syncslicemap_test.go]: changed too

**Review by henrybarreto — APPROVED** [review]

**Review by heiytor — APPROVED** [review]

**Pull request merged**